packages/aws-rfdk/lib/deadline/lib/spot-event-plugin-fleet.ts

```diff
@@ -0,0 +1,627 @@
+      return subnet.subnetId;
+    });
+    const subnetId = subnetIds.length != 0 ? subnetIds.join(',') : undefined;
+
+    const instanceTagsToken = this.tagsToken(SpotFleetResourceType.INSTANCE);
+    const spotFleetRequestTagsToken = this.tagsToken(SpotFleetResourceType.SPOT_FLEET_REQUEST);
+
+    const launchSpecifications: SpotFleetRequestLaunchSpecification[] = [];
+
+    props.instanceTypes.map(instanceType => {
+      const launchSpecification: SpotFleetRequestLaunchSpecification = {
+        blockDeviceMappings,
+        iamInstanceProfile: {
+          arn: iamProfile.attrArn,
+        },
+        imageId: this.imageId,
+        keyName: props.keyName,
+        securityGroups: securityGroupsToken,
+        subnetId,
+        tagSpecifications: instanceTagsToken,
+        userData: userDataToken,
+        instanceType: instanceType.toString(),
+      };
+      launchSpecifications.push(launchSpecification);
+    });
+
+    const spotFleetRequestProps: SpotFleetRequestProps = {
+      allocationStrategy: props.allocationStrategy ?? SpotFleetAllocationStrategy.LOWEST_PRICE,
+      iamFleetRole: this.fleetRole.roleArn,
+      launchSpecifications: launchSpecifications,
+      replaceUnhealthyInstances: props.replaceUnhealthyInstances ?? true,
+      // In order to work with Deadline, the 'Target Capacity' of the Spot fleet Request is
+      // the maximum number of Workers that Deadline will start.
+      targetCapacity: props.maxCapacity,
+      terminateInstancesWithExpiration: props.terminateInstancesWithExpiration ?? true,
+      // In order to work with Deadline, Spot Fleets Requests must be set to maintain.
+      type: SpotFleetRequestType.MAINTAIN,
+      validUntil: props.validUntil ? props.validUntil?.date.toUTCString() : undefined,
+      tagSpecifications: spotFleetRequestTagsToken,
+    };
+
+    const spotFleetRequestConfigurations = props.deadlineGroups.map(group => {
+      const spotFleetRequestConfiguration: SpotFleetRequestConfiguration = {
+        [group]: spotFleetRequestProps,
+      };
+      return spotFleetRequestConfiguration;
+    });
+
+    return spotFleetRequestConfigurations;
+  }
+
+  private tagsToken(resourceType: SpotFleetResourceType): IResolvable {
+    return Lazy.any({
+      produce: () => {
+        if (this.tags.hasTags()) {
+          const tagSpecification: SpotFleetTagSpecification = {
+            resourceType: resourceType,
+            tags: this.tags.renderTags(),
+          };
+          return [tagSpecification];
+        }
+        return undefined;
+      },
+    });
+  }
+
+  private validateProps(props: SpotEventPluginFleetProps): void {
+    this.validateInstanceTypes(props.instanceTypes);
+    this.validateGroups('deadlineGroups', props.deadlineGroups);
+    this.validateRegion('deadlineRegion', props.deadlineRegion);
+    this.validateBlockDevices(props.blockDevices);
+  }
+
+  private validateInstanceTypes(array: InstanceType[]): void {
+    if (array.length == 0) {
```

Comment:
Nitpicking again, but wouldn't the strict `===` operator be better?

Reply:
Sure, thanks for noticing!